src/coins.cpp

```diff
@@ -220,12 +234,29 @@ bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn
 }
 
 bool CCoinsViewCache::Flush() {
-    bool fOk = base->BatchWrite(cacheCoins, hashBlock);
+    bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase*/ true);
```

Comment:
The following `TODO` comment in `src/validation.cpp` may be removed in #23738 (I will edit/update this comment if it is removed). Making a note here so the `TODO` can be either be done in this PR or in a follow-up.
```
    coins_cache.Flush(); // TODO: if #17487 is merged, add erase=false here for better performance.
```

Reply:
Note: I found that the tests don't fail if `erase` is flipped to `false` on this line, other than in `Sync()`. The tests would need to become more granular to catch this case but it certainly has not gotten worse than from before this PR so it can be kept for a follow-up if it seems beneficial to test.